tests/assemblers/test_xgboost.py

```diff
@@ -352,17 +352,17 @@ def test_regression_random_forest():
             ast.IfExpr(
                 ast.CompExpr(
                     ast.FeatureRef(5),
-                    ast.NumVal(6.94099998),
+                    ast.NumVal(6.8410000801),
                     ast.CompOpType.GTE),
-                ast.NumVal(18.1008453),
-                ast.NumVal(9.60167599)),
+                ast.NumVal(17.4066162109),
+                ast.NumVal(9.6789960861)),
             ast.IfExpr(
                 ast.CompExpr(
-                    ast.FeatureRef(5),
-                    ast.NumVal(6.79699993),
+                    ast.FeatureRef(12),
```

Comment:
Not sure I understand why the feature index changed here.

Reply:
Oh, all tests have changed due to new train/test splitting routine which is now done by scikit-learn function (`train_test_split`). I found it easier to change `random_state` param only in one place compared to manual `shuffle` in multiple places.

Reply:
Oh, stupid me. I overlooked that change. It's a bit harder to compare apples to apples because of this though. Do you by chance remember whether there were any changes to expected values before you updated the splitting logic? I'm trying to identify the impact of this update.

Reply:
Sorry, updating splitting logic was the first step as I wanted to play around with `random_state`. Let's choose easier way. Let me split this PR into two.

Reply:
Done in aeb4301dfc88c85edd79e7a8d1304d96a3d35ba9.

Reply:
Thank you, this is so much better!